juju/juju
# feat: double-write unit states

This wires up the `unitstate` domain service to the `uniter` API and uses it to write unit hook states to Dqlite in addition to Mongo. This is a significant addition, because it adds our hottest write transaction path to Dqlite.

## QA steps

- Bootstrap to LXD and add 3 units of `postgresql`.
- `make repl-install`
- s
- `select * from unit_state` should yield something like:
```
unit_uuid                               uniter_state    storage_state   secret_state
04829d08-6e53-4a00-878a-4776507c67c3    leader: true
started: true
stopped: false
installed: true
removed: false
status-set: true
op: continue
opstep: pending
config-hash: 4d9d5e9f38933cf71dfa0c989b824b65ab606f397fc556a1e142704140a69cf0
trust-hash: 18cbe02213f17ac8ed6e4e9a5f7636aeb4767855001f64053217a2e5ee21ceb6
addresses-hash: 71b9b39bc27eb9a3bbdb340d52227c980c46b1e455c9c7b23a672c0f3ff5767f
        pgdata/2: true
        secret-obsolete-revisions:
  crlalcjm7h3i1phq0a00:
  - 1

9da332c1-d910-4929-8a7b-829658d002f4    leader: false
started: true
stopped: false
installed: true
removed: false
status-set: true
op: continue
opstep: pending
config-hash: 4d9d5e9f38933cf71dfa0c989b824b65ab606f397fc556a1e142704140a69cf0
trust-hash: 18cbe02213f17ac8ed6e4e9a5f7636aeb4767855001f64053217a2e5ee21ceb6
addresses-hash: 27b6e6cac720086f22bc102060937f580dd24cc0a2246d595579c8862173dddd
                                        pgdata/1: true
                                        <nil>
e4b5ccc4-8406-4685-86c7-067740aadc96    leader: false
started: true
stopped: false
installed: true
removed: false
status-set: true
op: continue
opstep: pending
config-hash: 4d9d5e9f38933cf71dfa0c989b824b65ab606f397fc556a1e142704140a69cf0
trust-hash: 18cbe02213f17ac8ed6e4e9a5f7636aeb4767855001f64053217a2e5ee21ceb6
addresses-hash: ed5c9888636527f76023e5569ec7ef11d4722852e04887ce09553b23cf22bdd4
        pgdata/0: true
        <nil>
```

## Documentation changes

None.

## Links

**Jira card:** [JUJU-6706](https://warthogs.atlassian.net/browse/JUJU-6706)
